react/features/transcription/middleware.js

```diff
@@ -0,0 +1,117 @@
+import { MiddlewareRegistry } from '../base/redux';
+
+import { ENDPOINT_MESSAGE_RECEIVED } from './actionTypes';
+import {
+    addTranscriptMessage,
+    removeTranscriptMessage,
+    updateTranscriptMessage
+} from './actions';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+* Time after which the rendered subtitles will be removed.
+*/
+const REMOVE_AFTER_MS = 3000;
+
+/**
+ * Middleware that catches actions related to transcript messages
+ * to be rendered in {@link TranscriptionSubtitles }
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+
+    switch (action.type) {
+    case ENDPOINT_MESSAGE_RECEIVED:
+        return _endpointMessageReceived(store, next, action);
+    }
+
+    return next(action);
+});
+
+/**
+ * Notifies the feature transcription that the action
+ * {@code ENDPOINT_MESSAGE_RECEIVED} is being dispatched within a specific redux
+ * store.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {Dispatch} next - The redux {@code dispatch} function to
+ * dispatch the specified {@code action} to the specified {@code store}.
+ * @param {Action} action - The redux action {@code ENDPOINT_MESSAGE_RECEIVED}
+ * which is being dispatched in the specified {@code store}.
+ * @private
+ * @returns {Object} The value returned by {@code next(action)}.
+ */
+function _endpointMessageReceived({ dispatch, getState }, next, action) {
+    const payload = action.payload;
+
+    try {
```

Comment:
this does not currently parse the new json-message structure correctly, right?

Reply:
Forgot to make a PR to your fork. After the json-message structure changes, we will be receiving the entire json here, not a separate payload. I'll change `payload` to `json`.